controller/src/main/java/org/jboss/as/controller/operations/global/ReadAttributeHandler.java

```diff
@@ -183,8 +183,26 @@ private void doExecuteInternal(OperationContext context, ModelNode operation) th
                 }
             }
             if (useEnhancedSyntax) {
-                ModelNode resolved = EnhancedSyntaxSupport.resolveEnhancedSyntax(attributeExpression.substring(attributeName.length()), context.getResult());
-                context.getResult().set(resolved);
+                // remove attribute name from expression string ("attribute-name.rest" => "rest")
+                int prefixLength = attributeName.length();
+                if (attributeExpression.length() > prefixLength && attributeExpression.charAt(prefixLength) == '.') {
```

Comment:
this looks bit redundant, as attributeExpression will always be longer as name attribute itself.
also if is of same size which would mean it is not expression at all, it will not come to this code block at all.